src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,834 @@
+  }, { once: true });
+}
+
+function showPanel(browser, ui, notificationId, telemetryId) {
+  let doc = browser.ownerDocument;
+  let panel = doc.defaultView.PopupNotifications.panel;
+
+  let populatePanel = (event) => {
+    if (event !== "shown") {
+      return;
+    }
+    let n = doc.getElementById(notificationId + "-notification");
+    let body = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-body");
+    let box = body.querySelector(".blurtsbox");
+    if (box) {
+      box.remove();
+    }
+    box = ui.box;
+    box.setAttribute("class", "blurtsbox");
+    box.setAttribute("style", "font-size: 110%");
+    body.appendChild(box);
+    let icon = doc.getAnonymousElementByAttribute(n, "class", "popup-notification-icon");
+    if (icon) {
+      icon.remove();
+    }
+  };
+
+  doc.defaultView.PopupNotifications.show(
+    browser, notificationId, "",
+    null, ui.primaryAction, ui.secondaryActions, {persistent: true, hideClose: true, eventCallback: populatePanel});
+  FirefoxMonitor.notifyEventListeners(`${telemetryId}_shown`);
+}
+
+function makeSpanWithLinks(aStrParts, doc) {
+  let spanElt = doc.createElementNS(HTML_NS, "span");
+  for (let str of aStrParts) {
+    if (!str.link) {
+      spanElt.appendChild(doc.createTextNode(str.str));
+      continue;
+    }
+    let anchor = doc.createElementNS(HTML_NS, "a");
+    anchor.setAttribute("style", "color: #0060DF");
+    anchor.setAttribute("href", "javascript:void(0)");
+    anchor.addEventListener("click", (event) => {
+      doc.defaultView.openUILinkIn(str.link, "tab", {});
+    });
+    anchor.appendChild(doc.createTextNode(str.str));
+    spanElt.appendChild(anchor);
+  }
+  return spanElt;
+}
+
+const XUL_NS = "http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul";
+const HTML_NS = "http://www.w3.org/1999/xhtml";
+let UIFactory = [
+
+  //variant #1
+  function(browser, doc, host, site) {
+    return {
+      get box() {
+        let box = doc.createElementNS(XUL_NS, "vbox");
+        let elt = doc.createElementNS(XUL_NS, "hbox");
+        elt.setAttribute("style", "margin-bottom: 1rem;");
+        elt.setAttribute("pack", "center");
+        let imgElt = doc.createElementNS(XUL_NS, "image");
+        imgElt.setAttribute("src", imageDataURIs.iconLaptopRisk);
+        imgElt.setAttribute("style", "width: 148px;");
+        elt.appendChild(imgElt);
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.setAttribute("anonid", "maindesc");
+        elt.appendChild(doc.createTextNode("Have an account on this website?\nIt may be at risk."));
+        elt.setAttribute("style", "font-size: 150%; text-align: center; white-space: pre; margin-bottom: 1rem;");
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        let strings = [
+          {str: site.Title + " was reported to "},
+          {str: "Firefox Monitor"},
```

Comment:
Should this be linked?
See L347 below.

Reply:
I wondered the same thing but according to https://docs.google.com/presentation/d/1udCP6jp1RUAxCpverqU4N6TFdVJi50e_MaIkJSXIuaE/edit#slide=id.p, it doesn't look like it is.